src/summary.rs

```diff
@@ -40,41 +41,57 @@ impl std::fmt::Display for Rewards {
     }
 }
 
-/// Struct for holding the information of what to be displayed with the `info`
-/// command
-#[derive(Deserialize, Serialize, Debug)]
+impl std::ops::AddAssign for Rewards {
+    fn add_assign(&mut self, other: Rewards) {
+        self.0 += other.0;
+    }
+}
+
+/// struct for flexibly updating the fields of the summary
+#[derive(Default, Debug)]
+pub(crate) struct SummaryUpdateFields {
+    pub(crate) is_plotting_finished: bool,
+    pub(crate) maybe_authored_count: Option<u64>,
+    pub(crate) maybe_vote_count: Option<u64>,
+    pub(crate) maybe_reward: Option<Rewards>,
+    pub(crate) maybe_new_blocks: Option<BlockNumber>,
+}
+
+/// Struct for holding the info of what to be displayed with the `info` command,
+/// and printing rewards to user in `farm` command
+#[derive(Deserialize, Serialize, Default, Debug, Clone, Copy)]
 pub(crate) struct Summary {
     pub(crate) initial_plotting_finished: bool,
-    pub(crate) farmed_block_count: u64,
+    pub(crate) authored_count: u64,
     pub(crate) vote_count: u64,
     pub(crate) total_rewards: Rewards,
     pub(crate) user_space_pledged: ByteSize,
+    pub(crate) last_processed_block_num: BlockNumber,
 }
 
-#[derive(Default, Debug)]
-pub(crate) struct SummaryUpdateFields {
-    pub(crate) is_plotting_finished: bool,
-    pub(crate) is_new_block_farmed: bool,
-    pub(crate) is_new_vote: bool,
-    pub(crate) maybe_new_reward: Option<Rewards>,
+impl Summary {
+    #[instrument]
+    pub(crate) async fn new(summary_file: SummaryFile) -> Result<Summary> {
+        summary_file.parse().await
+    }
 }
 
 /// utilizing persistent storage for the information to be displayed for the
 /// `info` command
 #[derive(Debug, Clone)]
-pub(crate) struct SummaryFilePointer {
-    file: Arc<Mutex<PathBuf>>,
+pub(crate) struct SummaryFile {
+    inner: Arc<Mutex<PathBuf>>,
```

Comment:
I think you either need to have `Arc<Mutex<File>>` here or just a `PathBuf`. `PathBuf` is not modified at all. I would prefer having wrapped `File` here

Reply:
The reason for `Arc<Mutex<PathBuf>>` was to measure against data races. There are different threads/tasks trying to reach this. 

I see your point, `PathBuf` itself is not being modified. However, by keeping it in a `Mutex`, we are giving the `key` to summary to only one thread/task. 

So, `PathBuf` alone (without Mutex) would not be safe. 
And I didn't want to put `File` in there, since I don't want to keep file open.

Reply:
> since I don't want to keep file open

Why so? I think it is should be fine

Reply:
because we have `info` command which can be run simultenaously whilst running a farmer. And `info` command reads this summary file